fe2-android/app/src/main/res/layout/settings_fragment.xml

```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+  xmlns:tools="http://schemas.android.com/tools">
+
+  <data>
+
+    <import type="android.view.View" />
+
+    <variable
+      name="viewmodel"
+      type="com.github.dedis.popstellar.ui.settings.SettingsViewModel" />
+
+    <variable
+      name="newUrl"
+      type="String" />
+  </data>
+
+  <androidx.constraintlayout.widget.ConstraintLayout
+    android:id="@+id/fragment_settings"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <LinearLayout
+      android:layout_width="match_parent"
+      android:layout_height="match_parent"
+      android:layout_marginHorizontal="@dimen/margin_text"
+      android:orientation="vertical">
+
+      <TextView
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/margin_top"
+        android:gravity="start"
+        android:text="@string/server_url"
+        android:textSize="@dimen/size_title"
+        android:textStyle="bold" />
+
+      <EditText
+        android:id="@+id/entry_box_server_url"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/margin_top"
+        android:autofillHints=""
+        android:hint="@string/text_hint_server_url"
+        android:inputType="textUri"
+        android:text="@{viewmodel.getServerUrl().toString()}"
+        android:afterTextChanged="@{(editable) -> viewmodel.setServerUrl(editable.toString())}" />
+
+      <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_marginHorizontal="@dimen/margin_button"
+        android:gravity="bottom|end"
+        android:orientation="horizontal"
+        tools:layout_editor_absoluteX="16dp">
+
+        <Button
+          android:id="@+id/button_apply"
+          style="@style/Theme.AppCompat"
+          android:layout_width="wrap_content"
+          android:layout_height="wrap_content"
+          android:enabled="@{entryBoxServerUrl.text.length() > 0 &amp;&amp; entryBoxServerUrl.text.toString() != viewmodel.getTempServerUrl()}"
```

Comment:
When we edit the url (without apply) then undo, the apply button is still enabled.
Maybe it's a problem with string "a != b", it seems that using !a.equals(b) works.
```suggestion
          android:enabled="@{entryBoxServerUrl.text.length() > 0 &amp;&amp; !entryBoxServerUrl.text.toString().equals( viewmodel.getTempServerUrl())}"
```